napari/layers/base/base.py

```diff
@@ -130,6 +133,23 @@ def __init__(
         if name is None and data is not None and os.getenv('MAGICNAME'):
             name = magic_name(data, path_prefix=ROOT_DIR)
 
+        # if we have a dask array, setup some sane defaults for optimized
+        # indexing and opportunistic caching.
+        if data is not None and isinstance(data, da.Array):
```

Comment:
Can you maybe also check if data is a list of dask arrays - i'm thinking about a multiscale use case here